src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveVirtualModifier.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+#nullable enable
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the 'virtual' modifier is not added to
+    /// or removed from a member.
+    /// </summary>
+    public class CannotAddOrRemoveVirtualModifier : Rule
+    {
+        public override void Initialize(RuleRunnerContext context) => context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+
+        private void RunOnMemberSymbol(ISymbol left, ISymbol right, ITypeSymbol leftContainingType, ITypeSymbol rightContainingType, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            // Members must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
+            // If the left member is virtual, it must not be sealed or inside a sealed type.
```

Comment:
```suggestion
            }

            // If the left member is virtual, it must not be sealed or inside a sealed type.
```
Style-NIT: Empty line after closing block. Same feedback on the rest of the cases.

Reply:
Also, I'm not sure I follow this part of the comment: `it must not be sealed or inside a sealed type.` I don't see how that relates to the code you write below.

Reply:
I mention the sealed thing because https://github.com/dotnet/sdk/issues/18682 originally mentions that the left member is overridable if it's virtual, not sealed, and its containing type is not sealed. However, it's not possible to declare a member with the virtual modifier that is also sealed or inside a sealed type, so the condition just reduces to `if (left.IsVirtual)`.

Reply:
right, I guess it's up to you, but I would just remove the seal comment since it isn't even possible to begin with. People won't look into the issue and instead look into the code only, and from the way the comment reads, I kind of understood from it that you were saying that if the ifcheck was true, then inside you would ensure the type was not sealed but that is not possible from the compiler.